src/app/services/build/MockBuildService.ts

```diff
@@ -0,0 +1,9 @@
+import { Build } from "../../models/project";
+import { Observable } from "rxjs/Observable";
+import { BuildService } from "./BuildService";
+
+export class MockBuildService implements BuildService {
+    getBuilds(projectId: string): Observable<Build[]> {
+        throw new Error("Method not implemented.");
+    }
```

Comment:
Is this where you'd imagine using `in-memory-web-api` to return the mock data? or something different?

Reply:
Either that, or simply strong type the existing mock data available in the `mock` directory.

Reply:
Do you recommend defining types as classes or interfaces? e.g. given this JSON:

```json
  {
    "id": "brigade-29d38c7477ecee18e184b69bec354fc350605c51bc16d4dd2b6073",
    "name": "technosophos/coffeesnob",
    "repo": {
      "name": "github.com/technosophos/coffeesnob",
      "cloneURL": "https://github.com/technosophos/coffeesnob.git"
    },
    "defaultScript": "",
    "kubernetes": {
      "namespace": "default",
      "vcsSidecar": "",
      "buildStorageSize": "50Mi"
    },
    "github": {
      "baseURL": "",
      "uploadURL": ""
    },
    "secrets": {},
    "worker": {
      "registry": "",
      "name": "",
      "tag": "",
      "pullPolicy": ""
    }
  }
```

```typescript
export class Project {
  constructor(
    public id: string,
    public name: string,
    public repo: Repo,
    public kubernetes: Kubernetes,
    public defaultScript: string,
    public github: any,
    public secrets: any,
    public worker: Worker
  ) {}
}
```

vs

```typescript
    export interface Project {
        id: string;
        name: string;
        repo: Repo;
        defaultScript: string;
        kubernetes: Kubernetes;
        defaultScript: string;
        github: Github;
        secrets: Secrets;
        worker: Worker;
    }
```